src/librustc/ty/error.rs

```diff
@@ -144,8 +144,7 @@ impl<'tcx> fmt::Display for TypeError<'tcx> {
                            found bound lifetime parameter {}", br)
             }
             Sorts(values) => ty::tls::with(|tcx| {
-                report_maybe_different(f, values.expected.sort_string(tcx),
-                                       values.found.sort_string(tcx))
+                write!(f, "expected {}", values.expected.sort_string(tcx))
```

Comment:
Looking through the unit tests that changed, I think I definitely underestimated the impact changing the label would have.

For the time being it looks like we'll need to keep it as-is, until we can come up with another solution.  If we removed the "expected" part, some of the labels just won't make sense or will become much more difficult to understand.